src/renderer/vanilla/util/render.js

```diff
@@ -49,18 +49,11 @@ const isTemplateElement = (element) => {
 		return false;
 	}
 
-	if (_cachedTemplateElements[tagName]) {
-		return true;
-	}
-
-	const elementClass = customElements.get(tagName);
-	const isTemplateElement = elementClass && elementClass._$templateElement$ === true;
-
-	if (isTemplateElement) {
-		_cachedTemplateElements[tagName] = elementClass;
+	if (element.shadowRoot) {
```

Comment:
@eddyloewen skuld we add this test to the other early return above as another OR case ?

Reply:
Yeah! Good point 👍

Reply:
Hold on... there is something fishy here... I changed the code but made a mistake but tests are still green :(
I'll have to check again tomorrow...

Reply:
Found a solution! I couldn't manage to write a failing test for the situation. But maybe it is not as important because the current vanilla renderer will be soon obsolete and for the new renderer this should not be a problem.